tracer/build/_build/Build.Steps.cs

```diff
@@ -586,9 +588,33 @@ void CopyWaf(string architecture, IEnumerable<string> frameworks, AbsolutePath a
             CompressZip(SymbolsDirectory, WindowsSymbolsZip, fileMode: FileMode.Create);
         });
 
+    Target ExtractDebugInfoAndStripSymbols => _ => _
+        .After(PublishProfilerLinux, PublishNativeLoaderLinux, PublishNativeProfilerLinux)
+        .OnlyWhenStatic(() => IsLinux)
+        .Executes(() =>
+        {
+            var directory = new DirectoryInfo(MonitoringHomeDirectory);
+            var files = directory.GetFiles("*.so");
+
+            var symbolsDirectory = SharedDirectory / "symbols";
+            EnsureExistingDirectory(symbolsDirectory);
+
+            foreach (var file in files)
+            {
+                var outputFile = Path.Combine(symbolsDirectory, Path.GetFileNameWithoutExtension(file.Name));
+
+                Logger.Info($"Extracting debug symbol for {file.FullName} to {outputFile}");
+                ExtractDebugInfo.Value(arguments: $"--only-keep-debug {file.FullName} {outputFile}.debug");
```

Comment:
Just as a sanity check, have we confirmed we can correctly _use_ these symbols as necessary?

Reply:
I did some test. The result is not ideal (thanks to the ELF format which forces us to have the binary used while taking a memory dump) but I have something where I can see our native symbols.